rct229/rulesets/ashrae9012019/section19/section19rule7.py

```diff
@@ -0,0 +1,347 @@
+        rmi_b = context.baseline
+        rmi_p = context.proposed
+
+        dict_of_zones_and_terminal_units_served_by_hvac_sys_b = (
+            get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmi_b)
+        )
+
+        hvac_system_serves_only_labs = True
+        are_any_lighting_space_types_defined = False
+        all_lighting_space_types_defined = True
+        for space_b in find_all(
+            "$.buildings[*].building_segments[*].zones[*].spaces[*]",
+            rmi_b,
+        ):
+            if hvac_system_serves_only_labs:
+                lighting_space_type_b = space_b.get("lighting_space_type")
+                if lighting_space_type_b is None:
+                    all_lighting_space_types_defined = False
+                else:
+                    are_any_lighting_space_types_defined = True
+                    if (
+                        lighting_space_type_b
+                        != LIGHTING_SPACE.LABORATORY_EXCEPT_IN_OR_AS_A_CLASSROOM
+                    ):
+                        hvac_system_serves_only_labs = False
+            else:
+                all_lighting_space_types_defined = False
+
+        zone_data = {}
+        for hvac_id_b in find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].id",
+            rmi_b,
+        ):
+            zone_data[hvac_id_b] = {
+                "zone_OA_CFM_list_of_schedules_b": [],
+                "zone_OA_CFM_list_of_schedules_p": [],
+                "was_DCV_modeled_baseline": False,
+                "was_DCV_modeled_proposed": False,
+                "zone_air_distribution_effectiveness_greater_than_1": False,
+            }
+            for zone_id_b in dict_of_zones_and_terminal_units_served_by_hvac_sys_b[
+                hvac_id_b
+            ]["zone_list"]:
+                zone_data[hvac_id_b]["zone_OA_CFM_list_of_schedules_b"].append(
+                    get_min_oa_cfm_sch_zone(rmi_b, zone_id_b)
+                )
+                zone_data[hvac_id_b]["zone_OA_CFM_list_of_schedules_p"].append(
+                    get_min_oa_cfm_sch_zone(rmi_p, zone_id_b)
+                )
+
+                zone_p = find_one(
+                    f'$.buildings[*].building_segments[*].zones[*][?(@.id = "{zone_id_b}")]',
+                    rmi_p,
+                )
+
+                zone_data[hvac_id_b]["was_DCV_modeled_baseline"] = any(
+                    [
+                        getattr_(
+                            terminal_p, "terminals", "has_demand_control_ventilation"
+                        )
+                        or find_one(
+                            f'$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*][?(@.id = "{getattr_(terminal_p, "Terminal", "served_by_heating_ventilating_air_conditioning_system")}")].fan_system.demand_control_ventilation_control',
+                            rmi_p,
+                        )
```

Comment:
This is incorrect - the flag is `was_DCV_modeled_baseline`, but `find_one` is searching proposed.
The RDS does not seem to require searching for demand_control_ventilation_control in baseline.
See line 56-57 in the RDS

Reply:
My apologies! It should be correct now.

Reply:
This seems still incorrect.
1. In RDS, [line56-57](https://github.com/pnnl/ruleset-checking-tool/blob/develop/docs/section19/Rule19-7.md?plain=1#L56-L57): The `was_DCV_modeled_baseline` is true if any terminal's `has_demand_control_ventilation` is true. It does not require checking the central fan system.